apps/omg_watcher_info/test/omg_watcher_info/db/eth_event_test.exs

```diff
@@ -418,6 +419,119 @@ defmodule OMG.WatcherInfo.DB.EthEventTest do
     )
   end
 
+  @tag fixtures: [:initial_blocks]
+  test "Allows for missing output when explicitly allowed" do
+    max_blknum = DB.Repo.aggregate(DB.TxOutput, :max, :blknum)
+    pos_from_future = Utxo.position(max_blknum + 1, 0, 0)
+
+    exits = [
+      %{
+        root_chain_txhash: Crypto.hash(<<6::256>>),
+        log_index: 0,
+        eth_height: 0,
+        call_data: %{utxo_pos: Utxo.Position.encode(pos_from_future)}
+      }
+    ]
+
+    assert :ok = DB.EthEvent.insert_exits!(exits, :in_flight_exit, false)
+  end
+
+  @tag fixtures: [:initial_blocks]
+  test "Fails when missing output disallowed" do
+    max_blknum = DB.Repo.aggregate(DB.TxOutput, :max, :blknum)
+    pos_from_future = Utxo.position(max_blknum + 1, 0, 0)
+
+    exits = [
+      %{
+        root_chain_txhash: Crypto.hash(<<6::256>>),
+        log_index: 0,
+        eth_height: 0,
+        call_data: %{utxo_pos: Utxo.Position.encode(pos_from_future)}
+      }
+    ]
+
+    assert_raise CaseClauseError, fn ->
+      DB.EthEvent.insert_exits!(exits, :in_flight_exit, true)
+    end
+  end
+
+  @tag fixtures: [:phoenix_ecto_sandbox, :alice]
+  test "deposited and exited utxo is retrievable by position", %{alice: alice} do
+    assert :ok =
+             DB.EthEvent.insert_deposits!([
+               %{
+                 root_chain_txhash: Crypto.hash(<<1000::256>>),
+                 eth_height: 1,
+                 log_index: 0,
+                 owner: alice.addr,
+                 currency: @eth,
+                 amount: 333,
+                 blknum: 1
+               }
+             ])
+
+    assert :ok =
+             DB.EthEvent.insert_exits!(
+               [
+                 %{
+                   root_chain_txhash: Crypto.hash(<<1001::256>>),
+                   eth_height: 2,
+                   log_index: 1,
+                   call_data: %{utxo_pos: Utxo.Position.encode(Utxo.position(1, 0, 0))}
+                 }
+               ],
+               :in_flight_exit,
+               true
+             )
+
+    assert %DB.TxOutput{ethevents: events} = DB.TxOutput.get_by_position(Utxo.position(1, 0, 0))
+
+    assert [
+             %DB.EthEvent{event_type: :deposit},
+             %DB.EthEvent{event_type: :in_flight_exit}
+           ] = Enum.sort(events, &(&1.eth_height < &2.eth_height))
+  end
+
+  @tag fixtures: [:initial_blocks]
+  test "only one exit type can be asssotiated to output which is retrievable by output_id",
```

Comment:
why only one?

Reply:
Because one exit event is enough to spend UTXO which is WI mostly awared of (and also only one effectively exits).